### DISCUSSION TITLE
(DAP) VSCode never requests variables from debug adapter

### DISCUSSION THREAD

**Discussion by smd21** [microsoft/vscode-discussions#1343] [Extension Development QnA] [answered]
I'm currently working on creating a debugger in VSCode using the[ Debug Adapter Protocol ](https://microsoft.github.io/debug-adapter-protocol/overview). After requesting the StackTrace and Scopes, VSCode just never makes any Variables requests despite several Scopes having variableReference >0 (which to my understanding should trigger a variables request). Is there something I missed in the DAP where the variables request is contingent on some other property? Really not sure what to do or how to diagnose what's going on.

**Comment by isidorn** [comment]
Some scopes are expensive and are not auto expanded. Make sure to double check that - it should be a property on the Scopes object. Also when you debug and use, when you expand the scopes manually - does VS Code then send a request?

Make sure to also read these docs https://code.visualstudio.com/api/extension-guides/debugger-extension

**Reply by smd21** [reply]
Yep I'm setting expensive to false. I'm using the Locals presentation hint (not certain if this is right though) and not using any of the other fields (namedVariables, indexedVariables, source, line, column, endLine, endColumn).  The UI doesn't show a way to expand the scopes

**Reply by isidorn** [reply]
Check the logs. If the UI shows scopes as not expandable something is off.
Check what you are doing differently compared to the mock debug adapter https://github.com/microsoft/vscode-mock-debug

**Reply by smd21** [reply]
I checked the logs but couldn't find anything regarding why the scopes are not shown as expandable. Do you know what files in particular may be helpful? I may have missed something.

 I'm handling scopes the same way as the mock debugger and have confirmed that the response does go through properly. Even when I was doing a mock case of just 1 scope it was still never showing as expandable. The package.json is laid out differently and I support a much smaller set of capabilities, would either of these be an issue?

**Reply by isidorn** [reply, marked as answer]
I think the best next step is to file an issue here https://github.com/microsoft/vscode/issues
With reproducable steps. Ideally you can share your Repository on GH so someone on the team can reproduce the problem you are seeing.
